src/Verilator.cpp

```diff
@@ -526,6 +526,54 @@ static void process() {
     // Note early return above when opt.cdc()
 }
 
+static void execBuildJob() {
+    if (!v3Global.opt.build()) return;
+
+    std::stringstream cmd;
+    const V3StringList& makeFlags = v3Global.opt.makeFlags();
+    const int jobs = v3Global.opt.buildJobs();
+    if (v3Global.opt.gmake()) {  // If both gmake and cmake are chosen, use gmake to build.
+        cmd <<  v3Global.opt.makeCmd();
+        cmd << " -C " << v3Global.opt.makeDir();
+        cmd << " -f " << v3Global.opt.prefix() << ".mk";
+        if (jobs == 0) {
+            cmd << " -j";
+        } else if (jobs > 1) {
+            cmd << " -j " << jobs;
+        }
+        for (V3StringList::const_iterator it = makeFlags.begin(); it != makeFlags.end(); ++it) {
+            cmd << ' ' << *it;
+        }
+    } else {
+        UASSERT(v3Global.opt.cmake(), "at least cmake or gmake must be chosen in V3Options.cpp");
+        cmd << "cd " << v3Global.opt.makeDir() << " && ";
+        cmd << "cmake";
+        for (V3StringList::const_iterator it = makeFlags.begin(); it != makeFlags.end(); ++it) {
+            cmd << ' ' << *it;
+        }
+        cmd << ' ' << V3Os::getcwd() << " && ";
+        cmd << "cmake --build . ";
+        if (jobs == 0) {
+            cmd << " -j";
+        } else if (jobs > 1) {
+            cmd << " -j " << jobs;
+        }
+    }
+    const std::string cmdStr = cmd.str();
+
+    const int ret = V3Os::system(cmdStr);
+    if (ret == -1) {
+        v3fatal("Failed to execute " << cmdStr);
```

Comment:
"Failed to execute make-system " ...

Reply:
Now the ret == -1 is checked in V3Os.cpp, because -1 means something was wrong in fork() or exec(), or waitpid(), ..... , but  not the command itself.

https://github.com/verilator/verilator/blob/593d7607d984b1396a5dc9da13bb9ce7997691dd/src/V3Os.cpp#L341-L343